### ISSUE TITLE
diwako_dui_compass_hide_blip_alone_group shows no player marker which there are objects in diwako_dui_special_track

### ISSUE THREAD

**Issue by Drofseh** [diwako/diwako_dui#59]
- Set `diwako_dui_compass_hide_blip_alone_group` to True
- Have no other squad members
- Add an object/unit to `diwako_dui_special_track`
- Object/unit blip appears in compass by player blip remains hidden
![](https://i.gyazo.com/cb39988371762934b9b027851e34aa3f.png)

This could be seen as expected behavior since the object in diwako_dui_special_track is, by design, never a member of the squad.
I think though that any time you have a blip for another object/unit then you should be able to see your own blip as well.

**Comment by diwako** [comment]
Hmm would require a rewrite for the player icon rendering part. Right now it is rendered simply as another unit tracked on the radar.

Not sure to be honest.

**Comment by Drofseh** [comment]
I think you could get away with having this check done sooner
https://github.com/diwako/diwako_dui/blob/master/addons/radar/functions/fnc_cacheLoop.sqf#L42

and then adding this
```sqf
if ((_specialTrack  != []) && (_group == []) then {
    _group = [player];
};
```
 inside this check
https://github.com/diwako/diwako_dui/blob/master/addons/radar/functions/fnc_cacheLoop.sqf#L16-L18

which would give 
```sqf
private _player = [] call CBA_fnc_currentUnit;
private _group = units group _player;
private _specialTrack = missionNamespace getVariable ["diwako_dui_special_track", []];
if (diwako_dui_compass_hide_blip_alone_group && {(count _group) <= 1}) then {
    _group = [];
    if ((_specialTrack  != []) && (_group == []) then {
        _group = [player];
    };
};
```

**Comment by Drofseh** [comment]
Any more thoughts on this @diwako ?

**Comment by diwako** [comment]
I haven't gotten around looking into it too much, life is busy atm. However vacation is coming up and that means I should have some free time to actually start into looking things.

**Comment by diwako** [comment]
@Drofseh gave it some thoughts and here we go
Result:
https://streamable.com/a87ym

611768de84df17e8208d1cd120971e2f02fc0e9d

**Comment by Drofseh** [comment]
Yah that works for me, thanks

**Comment by diwako** [comment]
Pushed to Dev workshop item for preview @Drofseh

**Comment by diwako** [comment]
Gonna reopen this as a reminder for my self, I need to add one more condition check or else people end up rendering your own icon twice, which is not wanted obviously:

Check for group size as well

**Comment by diwako** [comment]
Fixed with 54704270c9f0d2fa754f943af063850a8741e790 . again, pushed for preview to dev workshop item
